libtenzir/include/tenzir/plugin.hpp

```diff
@@ -640,13 +640,22 @@ class language_plugin : public virtual plugin {
 // request path.
 class rest_endpoint_plugin : public virtual plugin {
 public:
-  /// OpenAPI spec for the plugin endpoints.
+  /// OpenAPI description of the plugin endpoints.
   /// @returns A `tenzir::data` object that is a record containing entries for
   /// the `paths` element of an OpenAPI spec.
   [[nodiscard]] virtual auto
-  openapi_specification(api_version version = api_version::latest) const -> data
+  openapi_endpoints(api_version version = api_version::latest) const -> data
     = 0;
 
+  /// OpenAPI description of the schemas used by the plugin endpoints, if any.
+  /// @returns A `tenzir::data` object that is a record containing entries for
+  /// the `paths` element of an OpenAPI spec.
+  [[nodiscard]] virtual auto
+  openapi_schemas(api_version /*version*/ = api_version::latest) const
+    -> std::optional<data> {
+    return std::nullopt;
+  }
```

Comment:
This is one of the rare methods in `plugins.hpp` that isn't pure virtual - should we make it pure virtual and delegate that "optionalness" to all `rest_endpoint_plugin` implementations by returning that `std::nullopt`? Or would that be too much boilerplate code?

Reply:
We have quite a few methods with default implementations for plugins, especially for those plugin types that implement other plugin types. For example, the `store` plugin implements everything for `parser` and `printer` plugins.

One question I have: `data` is already nullable, so what's the difference between returning a `std::nullopt` and a `data{}` here?

Reply:
Imho `std::optional<data>` is explicitly saying "this may be empty" on a type level; but I've now switched to `record` so I guess the point is moot anyways.